In [188]:
import pandas as pd
import numpy as np
import seaborn as sns
import plotly.express as px
import plotly.io as pio
import plotly.graph_objects as go

# pio.renderers.default = "svg" # this line must be commented if working on colab

In [189]:
# !file "Speed+Dating+Data.csv"

In [190]:
# with open('Speed+Dating+Data.csv') as f:
#     print(f)

In [191]:
df = pd.read_csv("Speed+Dating+Data.csv", encoding='windows-1252')  # dérivé de encoding='latin-1'
df

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,...,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
0,1,1.0,0,1,1,1,10,7,NaN,4,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
1,1,1.0,0,1,1,1,10,7,NaN,3,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
2,1,1.0,0,1,1,1,10,7,NaN,10,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
3,1,1.0,0,1,1,1,10,7,NaN,5,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
4,1,1.0,0,1,1,1,10,7,NaN,7,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
8373,552,22.0,1,44,2,21,22,14,10.0,5,...,8.0,5.0,7.0,6.0,7.0,9.0,5.0,9.0,5.0,6.0
8374,552,22.0,1,44,2,21,22,13,10.0,4,...,8.0,5.0,7.0,6.0,7.0,9.0,5.0,9.0,5.0,6.0
8375,552,22.0,1,44,2,21,22,19,10.0,10,...,8.0,5.0,7.0,6.0,7.0,9.0,5.0,9.0,5.0,6.0
8376,552,22.0,1,44,2,21,22,3,10.0,16,...,8.0,5.0,7.0,6.0,7.0,9.0,5.0,9.0,5.0,6.0


In [192]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8378 entries, 0 to 8377
Columns: 195 entries, iid to amb5_3
dtypes: float64(174), int64(13), object(8)
memory usage: 12.5+ MB


In [193]:
# for i, col in enumerate(df.columns):
#     print(f"{i}: {col}")

In [194]:
# ajout colonne label genre et race

df["gender_label"] = df["gender"].map({0: "Femme", 1: "Homme"})
df["race_label"] = df["race"].map({1: "Noir", 2: "Blanc", 3:"Hispanique/Latino", 4: "Asiatique", 5: "Amérindien", 6: "Autre"})

In [195]:
# visualisation nb Hommes Femmes par race

fig_gender_race = px.histogram(df, x="race_label", color="gender_label")

fig_gender_race.update_layout(xaxis_title='Race', yaxis_title="Nombre")
fig_gender_race.update_layout(legend_title_text='Genre')
fig_gender_race.show()

- Comparer les couleurs ensemble (variables attributs), voir évolution dans le tps, y a-t-il des diffs selon revenus, études, religion, race, genre ?
- Voir les variables "uniques"

In [196]:
#a = len(df.loc[df["wave"].between(6, 9)].groupby("iid"))
#a

In [197]:
# Les vagues 6 à 9 ne vont pas être écartées car représentent 19% du dataset.

nb_participant_total = len(df.groupby('iid'))
nb_participant_w_6_9 = len(df.loc[(df["wave"] >= 6) & (df["wave"] <= 9)].groupby("iid")) # len(df.loc[df["wave"].between(6, 9)].groupby("iid"))
print(f"nb de participants : {nb_participant_total}, nb de participants vagues 6 à 9 : {nb_participant_w_6_9}, vagues 6 à 9 représentent {nb_participant_w_6_9/nb_participant_total:.2%} du dataset")

nb de participants : 551, nb de participants vagues 6 à 9 : 102, vagues 6 à 9 représentent 18.51% du dataset


In [198]:
# Test pour hamoniser les vagues 6 à 9. 6 à 60 si x10, 1 => 0 et 10 => 9, amplitude 10 - 1 = 9

six_att = [5, 7, 8, 1, 6, 8]
decal = [x - 1 for x in six_att]
print(decal)
rescal = [x * 100 / 9 for x in decal]
print(rescal)
somme = sum(rescal)
print(somme)
renorm = [x * 100 / somme for x in rescal]
print(renorm)
print(f"somme : {sum(renorm)}")
arrond = [round(x, 2) for x in renorm]
print(arrond)
print(f"somme arrondie : {sum(arrond)}")

[4, 6, 7, 0, 5, 7]
[44.44444444444444, 66.66666666666667, 77.77777777777777, 0.0, 55.55555555555556, 77.77777777777777]
322.22222222222223
[13.793103448275861, 20.689655172413794, 24.137931034482758, 0.0, 17.24137931034483, 24.137931034482758]
somme : 100.0
[13.79, 20.69, 24.14, 0.0, 17.24, 24.14]
somme arrondie : 100.0


In [199]:
# les vagues 6 à 9 vont être harmonisées. min 6 à max 60 si x10, 1 => 0 et 10 => 9, amplitude 10 - 1 = 9

df1 = df.copy()

wav6_9 = df1["wave"].between(6, 9)
df1[wav6_9]

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,...,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3,gender_label,race_label
1846,132,1.0,0,1,1,6,5,2,2.0,5,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Femme,Blanc
1847,132,1.0,0,1,1,6,5,2,2.0,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Femme,Blanc
1848,132,1.0,0,1,1,6,5,2,2.0,3,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Femme,Blanc
1849,132,1.0,0,1,1,6,5,2,2.0,4,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Femme,Blanc
1850,132,1.0,0,1,1,6,5,2,2.0,1,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Femme,Blanc
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3403,233,20.0,1,40,2,9,20,20,11.0,10,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Homme,Asiatique
3404,233,20.0,1,40,2,9,20,1,11.0,11,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Homme,Asiatique
3405,233,20.0,1,40,2,9,20,7,11.0,17,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Homme,Asiatique
3406,233,20.0,1,40,2,9,20,10,11.0,20,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Homme,Asiatique


In [200]:
# Voir quelles colonnes normaliser, pour les vagues de 6 - 9.
# A norm : attr4_1, attr3_1, attr3_s, attr3_2, attr4_2, attr4_3, attr2_3, attr3_3, attr_0, attr 
# déjà norm :pf_o_att, attr1_1, attr2_1, attr1_s, attr1_2, attr2_2, attr1_3,
# absent : attr5_1, attr7_2 , attr5_2, attr7_3, attr5_3 

# autres vagues sur 100 : pf_o_att, attr1_1, attr2_1, att1_2, attr1_3
# 1 à 10 : attr_o, attr3_1, attr, attr3_2, attr3_3

# pareil : pf_o_att, attr_o, attr_1, attr2_1, attr3_1, 

# => il n'y aurait que ces 5x6-2 colonnes à renormaliser, les autres étant notées sur la même échelle, des erreurs ds dataset des notes de 12/10 par ex... attr3_3, ligne 3130...
# A norm : attr4_1, attr3_s, attr4_2, attr4_3, attr2_3, 

verif_val = ["pf_o_att", "attr1_1", "attr4_1", 'attr2_1', 'attr3_1', 'attr5_1', "attr_o", "attr", "attr1_s", "attr3_s", "attr7_2", "attr1_2", "attr4_2", "attr2_2",
              "attr3_2", "attr5_2", "attr1_3", "attr7_3", "attr4_3", "attr2_3", "attr3_3", "attr5_3"]

# print(df1[verif_val].count(), )
# print(wav6_9[verif_val].count())

In [201]:
# Renormaliser colonnes vagues 6-9

col_norm = [["attr4_1", "sinc4_1", "intel4_1", "fun4_1", "amb4_1", 'shar4_1'], 
            ["attr4_2", "sinc4_2", "intel4_2", "fun4_2", "amb4_2", "shar4_2"],
            ["attr4_3", "sinc4_3", "intel4_3", "fun4_3", "amb4_3", "shar4_3"],
            ["attr2_3", "sinc2_3", "intel2_3", "fun2_3", "amb2_3", "shar2_3"]]


for col_list in col_norm:
    for col in col_list :
        df1.loc[wav6_9, col] = (df1.loc[wav6_9, col] - 1) * 100 / 9

    for idx_ligne in df1[wav6_9].index:
        ligne_vals = df1.loc[idx_ligne, col_list]
        somme_ligne = sum(ligne_vals)
        df1.loc[idx_ligne, col_list] = ligne_vals * 100 / somme_ligne
        
for col in col_norm:
    df1.loc[wav6_9, col] = df1.loc[wav6_9, col].round(2)


df1[wav6_9]["attr3_2"]

1846    9.0
1847    9.0
1848    9.0
1849    9.0
1850    9.0
       ... 
3403    6.0
3404    6.0
3405    6.0
3406    6.0
3407    6.0
Name: attr3_2, Length: 1562, dtype: float64

In [202]:
# Correction des valeurs 0 en 1 pour respecter l'échelle de notation de 1 à 10.

df1[["attr3_3", "sinc3_3", "intel3_3", "fun3_3", "amb3_3", "attr_o","sinc_o","intel_o","fun_o","amb_o","shar_o","attr","sinc", "intel", "fun","amb", "shar"]].describe()

col_mask_1 = ["attr3_3", "sinc3_3", "intel3_3", "fun3_3", "amb3_3", "attr_o","sinc_o","intel_o","fun_o","amb_o","shar_o","attr","sinc", "intel", "fun","amb", "shar"]
mask_1 = df1[col_mask_1] <= 0
df1[col_mask_1] = df1[col_mask_1].mask(mask_1, 1)

df1[col_mask_1].min()

attr3_3     2.0
sinc3_3     2.0
intel3_3    3.0
fun3_3      2.0
amb3_3      1.0
attr_o      1.0
sinc_o      1.0
intel_o     1.0
fun_o       1.0
amb_o       1.0
shar_o      1.0
attr        1.0
sinc        1.0
intel       1.0
fun         1.0
amb         1.0
shar        1.0
dtype: float64

In [203]:
# Correction des valeurs 12, 11 et 10.5 en 10

df1[["attr3_3", "sinc3_3", "intel3_3", "fun3_3", "amb3_3", "attr_o", "fun_o"]].describe()

col_mask_12 = ["attr3_3", "sinc3_3", "intel3_3", "fun3_3", "amb3_3", "attr_o", "fun_o"]
mask_12 = df1[col_mask_12] > 10
df1[col_mask_12] = df1[col_mask_12].mask(mask_12, 10)

df1[col_mask_12].max()

attr3_3     10.0
sinc3_3     10.0
intel3_3    10.0
fun3_3      10.0
amb3_3      10.0
attr_o      10.0
fun_o       10.0
dtype: float64

In [204]:
# df1[["attr_o","sinc_o","intel_o","fun_o","amb_o","shar_o","attr","sinc", "intel",
#      "fun","amb", "shar","like", "prob", "met","dec"]].describe()

In [205]:
# Normalisation colonnes sur toutes les vagues

col_norm_2 = [["attr3_1", "sinc3_1", "intel3_1", "fun3_1", "amb3_1"],
              ["attr3_s", "sinc3_s", "intel3_s", "fun3_s", "amb3_s"],
              ["attr3_2", "sinc3_2", "intel3_2", "fun3_2", "amb3_2"],
              ["attr3_3", "sinc3_3", "intel3_3", "fun3_3", "amb3_3"],
              ["attr5_1", "sinc5_1", "intel5_1", "fun5_1", "amb5_1"],
              ["attr5_2", "sinc5_2", "intel5_2", "fun5_2", "amb5_2"],
              ["attr5_3", "sinc5_3", "intel5_3", "fun5_3", "amb5_3"],
              ["attr_o", "sinc_o", "intel_o", "fun_o", "amb_o", "shar_o"],
              ["attr", "sinc", "intel", "fun", "amb", "shar"]]

for col_list in col_norm_2:
    for col in col_list :
        df1.loc[:, col] = (df1.loc[:, col] - 1) * 100 / 9

    for idx_ligne in df1.index:
        ligne_vals = df1.loc[idx_ligne, col_list]
        somme_ligne = sum(ligne_vals)
        df1.loc[idx_ligne, col_list] = ligne_vals * 100 / somme_ligne
        
for col in col_norm_2:
    df1.loc[:, col] = df1.loc[:, col].round(2)

In [206]:
# df1.to_csv("Speed_Dating_Data.csv", index=False)

## Comparer les 6 attributs à travers les différents temps par homme et femme.

In [207]:
# Q1 : Ce que le participant recherche dans le sexe opposé ?

map_col_q1 = {
    "attr1_1": ("t0", "attractivité"), "attr1_s": ("t1", "attractivité"), "attr1_2": ("t2", "attractivité"), "attr1_3": ("t3", "attractivité"),
    "sinc1_1": ("t0", "sincérité"), "sinc1_s": ("t1", "sincérité"), "sinc1_2": ("t2", "sincérité"), "sinc1_3": ("t3", "sincérité"),
    "intel1_1": ("t0", "intelligence"), "intel1_s": ("t1", "intelligence"), "intel1_2": ("t2", "intelligence"), "intel1_3": ("t3", "intelligence"),
    "fun1_1": ("t0", "fun"),"fun1_s": ("t1", "fun"), "fun1_2": ("t2", "fun"), "fun1_3": ("t3", "fun"),
    "amb1_1": ("t0", "ambition"), "amb1_s": ("t1", "ambition"), "amb1_2": ("t2", "ambition"), "amb1_3": ("t3", "ambition"), 
    'shar1_1': ("t0", "intérêts partagés"), "shar1_s": ("t1", "intérêts partagés"), "shar1_2": ("t2", "intérêts partagés"), "shar1_3": ("t3", "intérêts partagés")
}

col_q1 = ["gender"] + list(map_col_q1.keys())

df_q1 = df1[col_q1].copy()
# df_q1
df_q1_long = df_q1.melt(id_vars=["gender"], var_name="colonne", value_name="valeur")
df_q1_long["temps"] = df_q1_long["colonne"].map(lambda x: map_col_q1[x][0])
df_q1_long["attributs"] = df_q1_long["colonne"].map(lambda x: map_col_q1[x][1])

# df_q1_long

In [208]:
df_moy_q1 = df_q1_long.groupby(['gender', 'temps', 'attributs'])['valeur'].mean().reset_index()
# df_moyennes
fig1 = px.line(df_moy_q1, x="temps", y="valeur", color="attributs", facet_col="gender", title="Ce que le participant recherche chez le sexe opposé ?,")
fig1.show()

### Pour les femmes
- La valeur la plus recherchée au temps 0 qui correspond aux attentes avant de s'inscrire à l'évènement est l'intelligence (21) suivi par la sincérité (18) et l'attractivité (18) et le fun(17), les 2 attentes les plus faibles sont les intérêts partagés (13) et l'ambition (13). 
- Au temps 1 qui correspond aux attentes à mi chemin pendant l'évènement, l'écart se réduit, toutes les variables diminuent sauf l'attractivité et les intérêtes partagés augmentent légèrement.
- Au temps 2, le lendemain de la participation à l'évènement, l'attractivié, le fun et les intérêts partagés progressent, pas de changement pour la sincérité, les attentes d'ambition diminue un peu.
- Au temps 3, 3 à 4 semaines après l'envoi des matches, les places s'affinent, l'attractivié (21) reste en tête suivi de près de l'intelligence (20), puis la sincérité (17) le fun (15), les intérêts partagés (14) et l'ambition (12).
### Pour les hommes
- t0, c'est l'attractivié qui prédomine (27), suivi par par l'intelligence (20), le fun (18), la sincérité (16), les intérêts partagés (11) et l'ambition (9);
- t1, même classement, juste les valeurs se recentrent, (24-18-15-14-11-10).
- t2, l'attractivité remonte en flèche (30), le fun (18) change sa place avec l'intelligence (17), les autres restent égales (15-12-8).
- t3, l'attractivité reste en tête (29), l'intelligence reprend sa place intiale (19), pour le reste rien ne change (17-16-1-9).
### Conclusion
L'écart est plus étalé chez les hommes que chez les femmes. Les variables intelligence, fun et sincérité se placent également considérés chez les 2 groupes. L'attractivité est plus importance pour les hommes, les intérêts partagés et l'ambition sont moins considérés des 2 côtés. Les attentes sont à peu près les mêmes, seuls les valeurs divergent.

In [209]:
# df_q1.max()

In [210]:
# Q2 : Ce que le participant pense que la plupart de ses collègues recherche chez le sexe opposé ?

map_col_q2 = {
    "attr4_1": ("t0", "attractivité"), "attr4_2": ("t2", "attractivité"), "attr4_3": ("t3", "attractivité"),
    "sinc4_1": ("t0", "sincérité"), "sinc4_2": ("t2", "sincérité"), "sinc4_3": ("t3", "sincérité"),
    "intel4_1": ("t0", "intelligence"), "intel4_2": ("t2", "intelligence"), "intel4_3": ("t3", "intelligence"),
    "fun4_1": ("t0", "fun"),"fun4_2": ("t2", "fun"), "fun4_3": ("t3", "fun"),
    "amb4_1": ("t0", "ambition"), "amb4_2": ("t2", "ambition"), "amb4_3": ("t3", "ambition"), 
    'shar4_1': ("t0", "intérêts partagés"), "shar4_2": ("t2", "intérêts partagés"), "shar4_3": ("t3", "intérêts partagés")
}

col_q2 = ["gender"] + list(map_col_q2.keys())

df_q2 = df1[col_q2].copy()
# df_q1
df_q2_long = df_q2.melt(id_vars=["gender"], var_name="colonne", value_name="valeur")
df_q2_long["temps"] = df_q2_long["colonne"].map(lambda x: map_col_q2[x][0])
df_q2_long["attributs"] = df_q2_long["colonne"].map(lambda x: map_col_q2[x][1])

df_moy_q2 = df_q2_long.groupby(['gender', 'temps', 'attributs'])['valeur'].mean().reset_index()
# df_moyennes
fig2 = px.line(df_moy_q2, x="temps", y="valeur", color="attributs", facet_col="gender", title="Ce que le participant pense que la plupart de ses collègues recherche chez le sexe opposé ?,")
fig2.show()

#### Au fil du temps les variables restent stables, les hommes (32) comme les femmes (27) pensent que l'attractivité est l'atribut le plus recherché 
#### chez la plupart de ses collègues, l'ambition est la variable la moins recherchée chez les hommes (10), pour les femmes (13). Les autres variables 
#### sont dans les 14 de moy, hors fun 19 chez les hommes, 17 ches les femmes.

In [211]:
# df_q2.max()

In [212]:
# Q3 : Ce que le participant pense que le sexe opposé recherche dans une rencontre ?

map_col_q3 = {
    "attr2_1": ("t0", "attractivité"), "attr2_2": ("t2", "attractivité"), "attr2_3": ("t3", "attractivité"),
    "sinc2_1": ("t0", "sincérité"), "sinc2_2": ("t2", "sincérité"), "sinc2_3": ("t3", "sincérité"),
    "intel2_1": ("t0", "intelligence"), "intel2_2": ("t2", "intelligence"), "intel2_3": ("t3", "intelligence"),
    "fun2_1": ("t0", "fun"),"fun2_2": ("t2", "fun"), "fun2_3": ("t3", "fun"),
    "amb2_1": ("t0", "ambition"), "amb2_2": ("t2", "ambition"), "amb2_3": ("t3", "ambition"), 
    'shar2_1': ("t0", "intérêts partagés"), "shar2_2": ("t2", "intérêts partagés"), "shar2_3": ("t3", "intérêts partagés")
}

col_q3 = ["gender"] + list(map_col_q3.keys())

df_q3 = df1[col_q3].copy()
# df_q1
df_q3_long = df_q3.melt(id_vars=["gender"], var_name="colonne", value_name="valeur")
df_q3_long["temps"] = df_q3_long["colonne"].map(lambda x: map_col_q3[x][0])
df_q3_long["attributs"] = df_q3_long["colonne"].map(lambda x: map_col_q3[x][1])

df_moy_q3 = df_q3_long.groupby(['gender', 'temps', 'attributs'])['valeur'].mean().reset_index()
# df_moyennes
fig3 = px.line(df_moy_q3, x="temps", y="valeur", color="attributs", facet_col="gender", title="Ce que le participant pense que le sexe opposé recherche dans une rencontre ?")
fig3.show()

#### Que ce soit pour les hommes ou les femmes, chacun a finalement été assez juste dans la vision de ce que l'autre sexe recherche, comparativement avec
#### le visuel précédent.

In [213]:
# df_q3.max()

In [214]:
# Q4 : Comment le participant se perçoit ?

map_col_q4 = {
    "attr3_1": ("t0", "attractivité"), "attr3_s": ("t1", "attractivité"), "attr3_2": ("t2", "attractivité"), "attr3_3": ("t3", "attractivité"),
    "sinc3_1": ("t0", "sincérité"), "sinc3_s": ("t1", "sincérité"), "sinc3_2": ("t2", "sincérité"), "sinc3_3": ("t3", "sincérité"),
    "intel3_1": ("t0", "intelligence"), "intel3_s": ("t1", "intelligence"), "intel3_2": ("t2", "intelligence"), "intel3_3": ("t3", "intelligence"),
    "fun3_1": ("t0", "fun"),"fun3_s": ("t1", "fun"), "fun3_2": ("t2", "fun"), "fun3_3": ("t3", "fun"),
    "amb3_1": ("t0", "ambition"), "amb3_s": ("t1", "ambition"), "amb3_2": ("t2", "ambition"), "amb3_3": ("t3", "ambition")
}

col_q4 = ["gender"] + list(map_col_q4.keys())

df_q4 = df1[col_q4].copy()
# df_q1
df_q4_long = df_q4.melt(id_vars=["gender"], var_name="colonne", value_name="valeur")
df_q4_long["temps"] = df_q4_long["colonne"].map(lambda x: map_col_q4[x][0])
df_q4_long["attributs"] = df_q4_long["colonne"].map(lambda x: map_col_q4[x][1])

df_moy_q4 = df_q4_long.groupby(['gender', 'temps', 'attributs'])['valeur'].mean().reset_index()
# df_moyennes
fig4 = px.line(df_moy_q4, x="temps", y="valeur", color="attributs", facet_col="gender", title="Comment le participant se perçoit ?")
fig4.show()

#### Les perceptions des participants fluctuent légèrement durant l'expérience, mais finalement reviennent à peu près aux valeurs intiales.
#### Le classement des attributs est similaire dans les 2 groupes hors l'intelligence en tête chez les hommes, suivi par la sincérité, équivalente chez les femmes.
#### Elles se perçoivent légèrement plus fun que les hô. Pour les 2 groupes l'ambition et l'attractivité figure en bas de classement.

In [215]:
# df1[["attr3_2", "sinc3_2", "intel3_2", "fun3_2", "amb3_2"]].describe()

In [216]:
#df_q4.max() # A faire, corriger les 12 en 10

In [217]:
# Q5 : Comment le participant pense que les autres le perçoivent ?

map_col_q5 = {
    "attr5_1": ("t0", "attractivité"), "attr5_2": ("t2", "attractivité"), "attr5_3": ("t3", "attractivité"),
    "sinc5_1": ("t0", "sincérité"), "sinc5_2": ("t2", "sincérité"), "sinc5_3": ("t3", "sincérité"),
    "intel5_1": ("t0", "intelligence"), "intel5_2": ("t2", "intelligence"), "intel5_3": ("t3", "intelligence"),
    "fun5_1": ("t0", "fun"), "fun5_2": ("t2", "fun"), "fun5_3": ("t3", "fun"),
    "amb5_1": ("t0", "ambition"), "amb5_2": ("t2", "ambition"), "amb5_3": ("t3", "ambition")
}

col_q5 = ["gender"] + list(map_col_q5.keys())

df_q5 = df1[col_q5].copy()
# df_q1
df_q5_long = df_q5.melt(id_vars=["gender"], var_name="colonne", value_name="valeur")
df_q5_long["temps"] = df_q5_long["colonne"].map(lambda x: map_col_q5[x][0])
df_q5_long["attributs"] = df_q5_long["colonne"].map(lambda x: map_col_q5[x][1])

df_moy_q5 = df_q5_long.groupby(['gender', 'temps', 'attributs'])['valeur'].mean().reset_index()
# df_moyennes
fig5 = px.line(df_moy_q5, x="temps", y="valeur", color="attributs", facet_col="gender", title="Comment le participant pense que les autres le perçoivent ?")
fig5.show()

#### Finalement de chaque côté le participant se perçoit comme il pense que les autres le perçoivent, peut-être difficile d'avoir suffisamment de recul sur soi-même.
Hormis les hommes qui pensent qu'on les perçoit moins sincères.

## Quels sont les attributs les moins recherchés chez un partenaire masculin ? Est-ce différent pour les partenaires féminines ?
#### à travers le temps :
plot1.png : Chez un partenaire les moins recherchés chez un partenaire masculin sont les intérêts partagés (14) et l'ambition (12) sur les 4 tps. pareil pour chez un partenaire féminin, ip (11) amb (9).
#### avec préférences partenaire déclarées :

In [218]:
df1[["pf_o_att", "pf_o_sin","pf_o_int","pf_o_fun","pf_o_amb","pf_o_sha","attr_o","sinc_o","intel_o","fun_o","amb_o","shar_o","attr","sinc", "intel",
     "fun","amb", "shar","like", "prob", "met","dec"]].describe()

,pf_o_att,pf_o_sin,pf_o_int,pf_o_fun,pf_o_amb,pf_o_sha,attr_o,sinc_o,intel_o,fun_o,...,attr,sinc,intel,fun,amb,shar,like,prob,met,dec
count,8289.000000,8289.000000,8289.000000,8280.000000,8271.000000,8249.000000,7019.000000,7019.000000,7019.000000,7019.000000,...,7028.000000,7028.000000,7028.000000,7028.000000,7028.000000,7028.000000,8138.000000,8069.000000,8003.000000,8378.000000
mean,22.495347,17.396867,20.270759,17.459714,10.685375,11.845930,15.495180,18.730969,19.485709,15.941027,...,15.494226,18.732992,19.485276,15.942403,17.414291,12.931000,6.134087,5.207523,0.948769,0.419909
std,12.569802,7.044003,6.782895,6.085526,6.126544,6.362746,4.820148,4.685446,4.316189,4.253220,...,4.821783,4.684934,4.316709,4.251943,4.822286,5.036837,1.841285,2.129565,0.989889,0.493573
min,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,15.000000,15.000000,17.390000,15.000000,5.000000,9.520000,13.040000,16.670000,17.020000,14.290000,...,13.040000,16.670000,17.020000,14.290000,15.000000,10.340000,5.000000,4.000000,0.000000,0.000000
50%,20.000000,18.370000,20.000000,18.000000,10.000000,10.640000,15.910000,18.180000,18.750000,16.670000,...,15.910000,18.180000,18.750000,16.670000,16.670000,13.790000,6.000000,5.000000,0.000000,0.000000
75%,25.000000,20.000000,23.810000,20.000000,15.000000,16.000000,18.180000,20.640000,21.050000,18.180000,...,18.180000,20.690000,21.050000,18.180000,19.230000,16.330000,7.000000,7.000000,2.000000,1.000000
max,100.000000,60.000000,50.000000,50.000000,53.000000,30.000000,55.560000,66.670000,66.670000,60.000000,...,55.560000,66.670000,66.670000,60.000000,70.000000,53.850000,10.000000,10.000000,8.000000,1.000000


In [219]:
part_pref = ["pf_o_att", "pf_o_sin","pf_o_int","pf_o_fun","pf_o_amb","pf_o_sha"]

map_attributs = {
    "pf_o_att": "attractivité",
    "pf_o_sin": "sincérité", 
    "pf_o_int": "intelligence",
    "pf_o_fun": "fun",
    "pf_o_amb": "ambition",
    "pf_o_sha": "intérêts partagés"
}

df1_part_pref = df1.melt(id_vars=["gender"], value_vars=part_pref, var_name="attribut", value_name="valeur")
# print(df1_part_pref)
df1_part_pref['attribut_clean'] = df1_part_pref['attribut'].map(map_attributs)
# print(df1_part_pref)
moy_per_att = df1_part_pref.groupby(["gender", "attribut_clean"])["valeur"].mean().reset_index()
# print(moy_per_att)
fig6 = px.pie(moy_per_att, names="attribut_clean", values="valeur", facet_col="gender", title="Préférences déclarées du partenaire au temps 1.")
fig6.show()

## Quelle importance les gens accordent-ils à l'attrait physique dans le choix d'un partenaire potentiel par rapport à son impact réel ?

In [220]:
df1_comparaison = df1.groupby(['gender', 'dec'])['attr'].mean().reset_index()
print(df1_comparaison)
fig7 = px.bar(df1_comparaison, x="dec", y="attr", color="gender", barmode="group", title="Quelle importance les gens accordent-ils à l'attrait physique ?")
fig7.show()

   gender  dec       attr
0       0    0  14.004194
1       0    1  16.026708
2       1    0  15.021916
3       1    1  17.438106


#### Il y a une petite importance accordée à l'attrait physique, pour les femmes 14 vs 16, pour les hommes 15 vs 17.5.

## Les intérêts communs sont-ils plus importants que l'origine ethnique commune ?

In [221]:
df1_acceptation = df1[df1["dec"]==1].copy()

df1_int_co_race = df1_acceptation.groupby(["gender"])[["int_corr", "samerace"]].mean().reset_index()
df1_int_co_race_long = df1_int_co_race.melt(id_vars=["gender"], value_vars=["int_corr", "samerace"], var_name="facteur", value_name="importance")

df1_int_co_race_long['gender_label'] = df1_int_co_race_long['gender'].map({0: 'Femmes', 1: 'Hommes'})
df1_int_co_race_long['facteur_label'] = df1_int_co_race_long['facteur'].map({'int_corr': 'Intérêts communs', 'samerace': 'Origine ethnique commune'})

# df1_int_co_race_long

fig8 = px.bar(df1_int_co_race_long, x="gender_label", y="importance", color="facteur_label", barmode="group", labels={"gender_label": "genres", "facteur_label": "facteurs"},
              title="Les intérêts communs sont-ils plus importants que l'origine ethnique ?")
fig8.show()

#### L'origine ethnique commune est plus importante que les intérêts communs dans la prise de décision.

## Les gens peuvent-ils prédire avec précision leur propre valeur perçue sur le marché des rencontres ?

#### Comme le montrent les fig 4 et 5, "Comment le participant se perçoit ?" et "Comment le participant pense que les autres le perçoivent ?", on retrouve une forme de cohérence dans leur auto-évaluation, le participant se perçoit comme il pense que les autres le perçoivent, et ce à travers la durée de l'expérience. Une légère nuance à appporter pour les hommes qui pensent qu'on les perçoit moins sincères qu'ils ne se perçoivent eux-mêmes.

In [222]:
# Q9 : Comment le partenaire perçoit le participant ?

map_col_q9 = {
    "attr_o": ("t1", "attractivité"),
    "sinc_o": ("t1", "sincérité"),
    "intel_o": ("t1", "intelligence"),
    "fun_o": ("t1", "fun"),
    "amb_o": ("t1", "ambition"),
    "shar_o": ("t1", "intérêts communs")
}

col_q9 = ["gender"] + list(map_col_q9.keys())
# print(col_q9)
df_q9 = df1[col_q9].copy()
# print(df_q9)
df_q9_long = df_q9.melt(id_vars=["gender"], var_name="colonne", value_name="valeur")
df_q9_long["attributs"] = df_q9_long["colonne"].map(lambda x: map_col_q9[x][1])

# eval auto perception
df_auto = df_q4_long[df_q4_long["temps"] == "t0"].groupby(['gender', 'attributs'])['valeur'].mean().reset_index()
df_auto["perspective"] = "Auto perception"

# eval supposée par les autres
df_supposee = df_q5_long[df_q5_long["temps"] == "t0"].groupby(['gender', 'attributs'])['valeur'].mean().reset_index()
df_supposee["perspective"] = "Perception supposée"

# eval réelle des partenaire
df_reelle = df_q9_long.groupby(['gender', 'attributs'])['valeur'].mean().reset_index()
df_reelle["perspective"] = "Perception réelle"

df_combined = pd.concat([df_auto, df_supposee, df_reelle], ignore_index=True)
df_combined
fig9 = px.bar(df_combined, x="attributs", y="valeur", color="perspective",barmode="group", facet_col="gender",
              title="Comparaison : Auto-perception vs Perception supposée vs Évaluation réelle")
fig9.show()

#### Les participants ont une capacité relativement correcte à prédire leur valeur perçue. Dans la perception réelle il y a une variable intérêts communs dans le calcul de la note sur 100 qui n'est pas présente dans l'auto perception et la perception supposée. En remettant ces valeurs dans les 2 autres groupes, les barres vertes augmenteraient ce qui réduirait l'écart.

## Pour obtenir un deuxième rendez-vous, vaut-il mieux être le premier ou le dernier rendez-vous de la soirée ?

In [223]:
premier_rdv = df1[df1["order"] == 1].copy()
# premier_rdv
dernier_rdv = df1.groupby("iid")["order"].max().reset_index()
dernier_rdv.columns = ["iid", "max_order"]
# dernier_rdv

df1_avec_max = df.merge(dernier_rdv, on="iid")
# df1_avec_max

dernier_rdv = df1_avec_max[df1_avec_max['order'] == df1_avec_max['max_order']].copy()
# dernier_rdv

print("=== RÉSULTATS ===")
print(f"Nombre de premiers rdv : {len(premier_rdv)}")
print(f"Nombre de derniers rdv : {len(dernier_rdv)}")

taux_premier = premier_rdv['dec'].mean()
taux_dernier = dernier_rdv['dec'].mean()

print(f"\nTaux de succès premier rdv : {taux_premier:.3f} ({taux_premier*100:.1f}%)")
print(f"Taux de succès dernier rdv : {taux_dernier:.3f} ({taux_dernier*100:.1f}%)")

difference = taux_dernier - taux_premier
print(f"\nDifférence : {difference:.3f} points")

if difference > 0:
    print("Le dernier rendez-vous semble plus favorable")
elif difference < 0:
    print("Le premier rendez-vous semble plus favorable") 
else:
    print("Pas de différence notable")

=== RÉSULTATS ===
Nombre de premiers rdv : 551
Nombre de derniers rdv : 551

Taux de succès premier rdv : 0.499 (49.9%)
Taux de succès dernier rdv : 0.452 (45.2%)

Différence : -0.047 points
Le premier rendez-vous semble plus favorable


## Visualisations pour expliquer pourquoi les gens acceptent un deuxième rendez-vous